### PR TITLE
Scaffold for mover

### PR DESCRIPTION
**Pull request by johandahlberg**
Prep-work for Mover. Ready for some initial code review, but since I don't know the details of how Mover will work I cannot really finish it.

### CODE REVIEW

**Comment by johandahlberg** [comment]
Added some stuff not directly related to the Mover things here, but to making it possible to stage any directory as a project,  see: https://snpseq.atlassian.net/browse/BIO-638 for more info.

File: delivery/repositories/project_repository.py [hunk 1]
```diff
@@ -21,3 +26,30 @@ def get_projects(self):
...
                 yield project
+
+
+class GeneralProjectRepository(object):
+    """
+    Repository for a general project. For this purpose a project is represented by any director in
+    root directory defined by the configuration.
+    """
+
+    def __init__(self, root_directory, filesystem_service=FileSystemService()):
```
Comment (johandahlberg) [inline delivery/repositories/project_repository.py]: To be able to easily mock it out in the tests.

File: delivery/handlers/delivery_handlers.py [hunk 2]
```diff
@@ -28,4 +28,12 @@ def post(self, staging_id):
         # TODO Extend this later once we know more about how this will work..
 
 
+class DeliveryStatusHandler(ArteriaDeliveryBaseHandler):
```
Comment (johandahlberg) [inline delivery/handlers/delivery_handlers.py]: :+1:

File: delivery/handlers/delivery_handlers.py [hunk 3]
```diff
@@ -28,4 +28,12 @@ def post(self, staging_id):
...
 
+    def initialize(self, **kwargs):
+        self.delivery_service = kwargs["delivery_service"]
+        super(DeliverByStageIdHandler, self).initialize(kwargs)
```
Comment (johandahlberg) [inline delivery/handlers/delivery_handlers.py]: :+1:

File: delivery/models/db_models.py [hunk 4]
```diff
@@ -81,6 +85,17 @@ class DeliveryOrder(SQLAlchemyBase):
...
+    # Optional path to md5sum file
+    md5sum_file = Column(String)
+
+    # Process id of Mover process used to start the delivery
+    mover_pid = Column(Integer)
+
+    # Mover delivery id - the id that is needed to query mover about
+    # a delivery status
+    # TODO At this stage I don't know the type of this... /JD 20161201
+    mover_delivery_id = Column(Integer)
```
Comment (johandahlberg) [inline delivery/models/db_models.py]: Good - so I'll probably go ahead and make this a String instead.

File: delivery/models/project.py [hunk 5]
```diff
@@ -3,15 +3,31 @@
...
+    """
+
+    def __eq__(self, other):
+        """
+        Two project should be considered the same if the represent the same directory on disk
+        :param other: instance of RunfolderProject
+        :return: true if the same project, otherwise false
+        """
+        if isinstance(other, self.__class__):
+            return self.path == other.path
```
Comment (johandahlberg) [inline delivery/models/project.py]: Also a good point, I think everywhere the paths are instantiated it's a absolute path, but I'll double check.

File: delivery/app.py [hunk 6]
```diff
@@ -39,15 +40,18 @@ def routes(**kwargs):
         url(r"/api/1.0/runfolders/(.+)/projects", ProjectsForRunfolderHandler,
             name="projects_for_runfolder", kwargs=kwargs),
 
-
         url(r"/api/1.0/stage/runfolder/(.+)", StagingRunfolderHandler,
             name="stage_runfolder", kwargs=kwargs),
+        url(r"/api/1.0/stage/project/(.+)", StageGeneralDirectoryHandler,
```
Comment (johanherman) [inline delivery/app.py]: I just meant that if you wanna follow semantic versioning then when you add something that breaks the interface, you bump arteria-delivery's version up to 2.0 from 1.0, and then preferably the routes should follow along.

**Comment by johandahlberg** [comment]
@johanherman if you think it's alright I think we could merge this and then continue work on coming things in another PR, before this one goes completely of the rails in size...

**Comment by johanherman** [comment]
Checking.

**Comment by johanherman** [comment]
:+1:

**Pull request merged**